js/ui/endlessButton.js

```diff
@@ -38,6 +39,14 @@ const EndlessButton = new Lang.Class({
             this._labelOffsetX = this._label.get_theme_node().get_length('-label-offset-x');
             this._labelOffsetY = this._label.get_theme_node().get_length('-label-offset-y');
         }));
+
+        Main.overview.connect('page-changed', Lang.bind(this, function() {
```

Comment:
Instead of setting the label from two different places, can you split the body of this function into its own private method, and call it from both the constructor and as a `page-changed` signal callback?

Also, I believe that the overview may outlive the button, so perhaps it would be a good idea to disconnect this signal when the button is destroyed.